Sources_v3/StreamChat/Controllers/SearchControllers/UserSearchController.swift

```diff
@@ -136,18 +144,55 @@ public class _ChatUserSearchController<ExtraData: ExtraDataTypes>: DataControlle
     }
     
     public func search(term: String, completion: ((_ error: Error?) -> Void)? = nil) {
-        var query = UserListQuery<ExtraData.User>(filter: .or([
-            .autocomplete(.name, text: term),
-            .autocomplete(.id, text: term)
-        ]))
-        query.filter?.explicitHash = explicitFilterHash
-        // We don't need to set `shouldBeObserved = false` since this is done when we first init the query
-        // with this filter hash (see `userListQueryUpdater` above)
-        
-        userQueryUpdater.update(userListQuery: query) { [weak self] error in
+        client.databaseContainer.write { [query] session in
+            session.removeAllUsers(from: query)
+        } completion: { [weak self] error in
             guard let self = self else { return }
-            self.state = error == nil ? .remoteDataFetched : .remoteDataFetchFailed(ClientError(with: error))
-            self.callback { completion?(error) }
+            if let error = error {
+                self.callback {
+                    completion?(error)
+                }
+                return
+            }
+            
+            var query = UserListQuery<ExtraData.User>(filter: .or([
+                .autocomplete(.name, text: term),
+                .autocomplete(.id, text: term)
+            ]))
+            query.filter?.explicitHash = self.explicitFilterHash
+            // We don't need to set `shouldBeObserved = false` since this is done when we first init the query
+            // with this filter hash (see `userListQueryUpdater` above)
+            
+            self.lastQuery = query
+            
+            self.userQueryUpdater.update(userListQuery: query) { [weak self] error in
+                guard let self = self else { return }
+                self.state = error == nil ? .remoteDataFetched : .remoteDataFetchFailed(ClientError(with: error))
+                self.callback { completion?(error) }
+            }
+        }
```

Comment:
I think this is unnecessary complex and also not that elegant as it can be :) Also, it has this ugly intermediate step when the previous results disappear before the new results are visible. My suggestion:
- move all this logic to `userQueryUpdater`
- introduce: 
```swift
enum MergeStrategy {
    case merge, replace
}
update(userListQuery: query, resultsMergeStrategy: .replace) { }
```
The code here will be nicer and more expressive + it will allow us to do everything in a single session.

What do you think?